pkg/kubectl/cmd/edit.go

```diff
@@ -0,0 +1,419 @@
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/editor"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/jsonmerge"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	//	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	//	"github.com/openshift/origin/pkg/cmd/util/editor"
+	//"github.com/openshift/origin/pkg/util/jsonmerge"
+)
+
+const (
+	editLong = `Edit a resource from the default editor.
+
+The edit command allows you to directly edit any API resource you can retrieve via the
+command line tools. It will open the editor defined by your OSC_EDITOR, GIT_EDITOR,
+or EDITOR environment variables, or fall back to 'vi'. You can edit multiple objects,
+although changes are applied one at a time. The command accepts filenames as well as
+command line arguments, although the files you point to must be previously saved
+versions of resources.
+
+The files to edit will be output in the default API version, or a version specified
+by --output-version. The default format is YAML - if you would like to edit in JSON
+pass -o json.
+
+In the event an error occurs while updating, a temporary file will be created on disk
+that contains your unapplied changes. The most common error when updating a resource
+is another editor changing the resource on the server. When this occurs, you will have
+to apply your changes to the newer version of the resource, or update your temporary
+saved copy to include the latest resource version.`
+
+	editExample = `  // Edit the service named 'docker-registry':
+  $ kubectl edit svc/docker-registry
+
+  // Edit the DeploymentConfig named 'my-deployment':
+  $ kubectl edit dc/my-deployment
+
+  // Use an alternative editor
+  $ OSC_EDITOR="nano" kubectl edit dc/my-deployment
+
+  // Edit the service 'docker-registry' in JSON using the v1beta3 API format:
+  $ kubectl edit svc/docker-registry --output-version=v1beta3 -o json`
+)
+
+var errExit = fmt.Errorf("exit directly")
+
+func NewCmdEdit(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	var filenames util.StringList
+	cmd := &cobra.Command{
+		Use:     "edit (RESOURCE/NAME | -f FILENAME)",
+		Short:   "Edit a resource on the server",
+		Long:    editLong,
+		Example: fmt.Sprintf(editExample),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunEdit(f, out, cmd, args, filenames)
+			if err == errExit {
+				os.Exit(1)
+			}
+			cmdutil.CheckErr(err)
+		},
+	}
+	usage := "Filename, directory, or URL to file to use to edit the resource"
+	kubectl.AddJsonFilenameFlag(cmd, &filenames, usage)
+	cmd.Flags().StringP("output", "o", "yaml", "Output format. One of: yaml|json.")
+	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
+	return cmd
+}
+
+func RunEdit(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, filenames util.StringList) error {
```

Comment:
@deads2k, I'm not sure if I follow you. Do you mean instead of passing in `cmd` and `factory`, we should pass in things like `cmdutil.GetFlagString(cmd, "output")`?